src/openldap/mod.rs

```diff
@@ -0,0 +1,519 @@
+///     openldap_instance.get_host_port_ipv4(1389).unwrap(),
+/// );
+/// let mut conn = ldap3::LdapConn::new(&connection_string).unwrap();
+/// let ldap3::SearchResult(rs,_) = conn.search(
+///              "ou=users,dc=example,dc=org",
+///              ldap3::Scope::Subtree,
+///              "(cn=ma*)",
+///              vec!["cn"]
+///          ).unwrap();
+/// let results:Vec<_>=rs.into_iter().map(ldap3::SearchEntry::construct).collect();
+/// assert_eq!(results.len(), 0);
+/// ```
+///
+/// [`OpenLDAP`]: https://www.openldap.org/
+/// [`bitnami/openldap docker image`]: https://hub.docker.com/r/bitnami/openldap
+/// [`OpenLDAP configuration guide`]: https://www.openldap.org/doc/admin26/guide.html
+#[derive(Debug, Clone)]
+pub struct OpenLDAP {
+    env_vars: HashMap<String, String>,
+    users: Vec<User>,
+}
+#[derive(Debug, Clone)]
+pub struct User {
+    username: String,
+    password: String,
+}
+
+impl OpenLDAP {
+    /// Sets the LDAP baseDN (or suffix) of the LDAP tree.
+    /// Default: `"dc=example,dc=org"`
+    pub fn with_base_dn(mut self, base_dn: &str) -> Self {
+        self.env_vars
+            .insert("LDAP_ROOT".to_owned(), base_dn.to_owned());
+        self
+    }
+    /// Sets an admin account (there can only be one)
+    /// Default username: `"admin"` => dn: `cn=admin,dc=example,dc=org` if using the default `base_dn` instead of overriding via [`OpenLDAP::with_base_dn`].
+    /// Default password: `"adminpassword"`
+    pub fn with_admin(mut self, username: &str, password: &str) -> Self {
+        self.env_vars
+            .insert("LDAP_ADMIN_USERNAME".to_owned(), username.to_owned());
+        self.env_vars
+            .insert("LDAP_ADMIN_PASSWORD".to_owned(), password.to_owned());
+        self
+    }
+
+    /// Sets a configuration admin user (there can only be one)
+    /// Default: `None`
+    pub fn with_config_admin(mut self, username: &str, password: &str) -> Self {
+        self.env_vars
+            .insert("LDAP_CONFIG_ADMIN_ENABLED".to_owned(), "yes".to_owned());
+        self.env_vars
+            .insert("LDAP_CONFIG_ADMIN_USERNAME".to_owned(), username.to_owned());
+        self.env_vars
+            .insert("LDAP_CONFIG_ADMIN_PASSWORD".to_owned(), password.to_owned());
+        self
+    }
+
+    /// Sets an accesslog admin user up (there can only be one)
+    /// Configuring the admin for the access log can be done via [`OpenLDAP::with_accesslog_admin`]
+    /// Default: `None`
+    pub fn with_accesslog_settings(
+        mut self,
+        AccesslogSettings {
+            log_operations,
+            log_success,
+            log_purge,
+            log_old,
+            log_old_attribute,
+        }: AccesslogSettings,
+    ) -> Self {
+        self.env_vars
+            .insert("LDAP_ENABLE_ACCESSLOG".to_owned(), "yes".to_owned());
+        let log_operations = match log_operations {
+            AccesslogLogOperations::Writes => "writes".to_owned(),
```

Comment:
"writes" in lowercase, but everything else is in capital letters. Is it expected? (case insensitive?)

Let's probably utilize [parse-display](https://crates.io/crates/parse-display) for the enum. This will reduce boilerplate and should be less error prone

Reply:
Good catch!
Everything here should be lowercase, will refactor using that dependency